runtime/commonTest/src/kotlinx/serialization/SerializersLookupEnumTest.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization
+
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.encoding.*
+import kotlinx.serialization.json.*
+import kotlinx.serialization.test.*
+import kotlin.test.*
+
+// This is unimplemented functionality that should be
+@Suppress("RemoveExplicitTypeArguments") // This is exactly what's being tested
+class SerializersLookupEnumTest : JsonTestBase() {
+    @Serializable(with = EnumExternalObjectSerializer::class)
+    enum class EnumExternalObject
+
+    @Serializer(forClass = EnumExternalObject::class)
+    object EnumExternalObjectSerializer {
+        override val descriptor: SerialDescriptor = buildSerialDescriptor("tmp", SerialKind.ENUM)
+
+        override fun serialize(encoder: Encoder, value: EnumExternalObject) {
+            TODO()
+        }
+
+        override fun deserialize(decoder: Decoder): EnumExternalObject {
+            TODO()
+        }
+    }
+
+    @Serializable(with = EnumExternalClassSerializer::class)
+    enum class EnumExternalClass
+
+    @Serializer(forClass = EnumExternalObject::class)
+    class EnumExternalClassSerializer {
+        override val descriptor: SerialDescriptor = buildSerialDescriptor("tmp", SerialKind.ENUM)
+
+        override fun serialize(encoder: Encoder, value: EnumExternalObject) {
+            TODO()
+        }
+
+        override fun deserialize(decoder: Decoder): EnumExternalObject {
+            TODO()
+        }
+    }
+
+    @Polymorphic
+    enum class EnumPolymorphic
+
+
+    @Test
+    fun testEnumExternalObject() {
+        assertFailsWith<SerializationException> { (serializer<EnumExternalObject>()) }
+    }
+
+    @Test
+    fun testEnumExternalClass() {
+        assertFailsWith<SerializationException> { serializer<EnumExternalClass>() }
+    }
+
+    @Test
+    fun testEnumPolymorphic() {
+        jvmOnly {
+            assertEquals(
```

Comment:
Is there any sense to provide and test such functionality? Does someone really uses polymorphic enums?

Reply:
No, but we are obliged to provide it. Otherwise, `serializer()` function will be just incorrect.
If we think this functionality is nonsense, we should prohibit it at the plugin level